src/vm/mod.rs

```diff
@@ -416,12 +430,13 @@ impl super::VM for CoreVM {
     }
 
     #[instrument(
-        level = "debug",
+        level = "trace",
         skip(self),
         fields(restate.invocation.id = self.debug_invocation_id(), restate.journal.index = self.context.journal.index(), restate.protocol.version = %self.version),
         ret
     )]
     fn sys_sleep(&mut self, duration: Duration) -> VMResult<AsyncResultHandle> {
+        invocation_debug_logs!(self, "Executing 'Sleep for {duration:?}'");
```

Comment:
the number that is printed, what does it represent? It seems very large in the log example you posted. 
Is it really the number of seconds that it will sleep?

Reply:
no this is the duration since unix epoch. Let me think how to solve this, without adding a "fat" library there is no easy way to print a date/time in rust.

Reply:
Could you not just do the inverse of what the TS wasm bindings is doing?
```
wake_up_time - duration_since_unix_epoch()
```
That will give a number of seconds until the wake up time I believe, so it still keeps the same logging format you have now (`'Sleep for 10s'`) rather than needing to show the timestamp of when it will sleep until

Reply:
> wake_up_time - duration_since_unix_epoch()

Not really, because this state machine is supposed to be pure/without side effects, it might not even have access to the system clock to begin with (in practice, I think SystemTime::now()  panics in webassembly)

Reply:
What do you think about this?

```
[restate][2024-12-13T14:22:35.566Z][greeter/greet][inv_12xNr8QTCBOx3Dl8GhXdjSSY5l1MEwNXxv] INFO: Starting invocation.
[restate][2024-12-13T14:22:35.573Z][greeter/greet][inv_12xNr8QTCBOx3Dl8GhXdjSSY5l1MEwNXxv] DEBUG: Executing 'Create awakeable'
[restate][2024-12-13T14:22:35.575Z][greeter/greet][inv_12xNr8QTCBOx3Dl8GhXdjSSY5l1MEwNXxv] INFO: curl -X POST http://localhost:8080/restate/awakeables/prom_1EbnQovICpR0Bk8BmYhm_lZ9BDjhCVmt3AAAAAQ/resolve --json '"Guardiani"'
[restate][2024-12-13T14:22:35.577Z][greeter/greet][inv_12xNr8QTCBOx3Dl8GhXdjSSY5l1MEwNXxv] DEBUG: Executing 'Sleeping up to 2024-12-13T14:22:45.577000000Z'
```

Reply:
essentially prints when the sleep terminates... Not super great honestly though...

Reply:
Ok i got another idea :)

Reply:
> [restate][2024-12-13T16:25:54.322Z][greeter/greet][inv_10QgwlVkaP8k6Wp5XwgsvFzf4bB9mSQAQp] DEBUG: Executing 'Sleeping for 10s'


Got it!